Assets/Scripts/Gameplay/Items/Item.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[RequireComponent(typeof(Collider))]
+public class Item : MonoBehaviour, IInteractable
+{
+    [SerializeField]
+    private ItemSkill m_ItemSkill;
+    public void Interact(ICanInteract interactor)
+    {
+        if (interactor != null && interactor is Player player)
+        {
+            PlayPickUpSound();
+            HandleItemSkill(player);
+            Destroy(gameObject);
+        }
+    }
+    
+    private void PlayPickUpSound()
+    {
+        AudioManager.m_Instance.PlaySound("MAGIC_Powerup", 1.0f, 1.2f);
+    }
+
+    public void HandleItemSkill(Player player)
+    {
+        Transform skillTransform = player.transform.Find("Skills");
+        
+        if (skillTransform == null)
+        {
+            Debug.LogError("Player should have a child GameObject named Skills");
+            return;
+        }
+
+        ItemSkill itemSkill = skillTransform.gameObject.AddComponent(m_ItemSkill.GetType()) as ItemSkill; //adds child type
+        itemSkill.InitializeSkill();
+        player.GetComponentInChildren<SkillHandler>().AddSkill(itemSkill); //must be after or else will be deleted
```

Comment:
You should do a null check for the component before using it

Reply:
I suppose we should just have a private skill handler field within player for this then. Then make a getter. Will guarantee existence in this case.

Reply:
Yeah but idk if the object is moved dynamically during runtime or not. Of course, if we can assign reference in the editor then do that instead 100%